### PR TITLE
Add gsf.py to lint checks and fix linting errors in file.

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

* Previously our lint check did not include top-level files under graphstorm/, this PR fixes that.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gsf.py [hunk 1]
```diff
@@ -116,17 +120,7 @@
...
-                   GSgnnMrrLPEvaluator)
-from .trainer import (GSgnnLinkPredictionTrainer,
-                      GSgnnNodePredictionTrainer,
-                      GSgnnEdgePredictionTrainer,
-                      GLEMNodePredictionTrainer)
-from .inference import (GSgnnLinkPredictionInferrer,
-                        GSgnnNodePredictionInferrer,
-                        GSgnnEdgePredictionInferrer,
-                        GSgnnEdgePredictionInferrer)
+                   GSgnnLPEvaluator,)
```
Comment (classicsong) [inline python/graphstorm/gsf.py:123]:
```suggestion
                   GSgnnLPEvaluator)
```

File: python/graphstorm/gsf.py [hunk 2]
```diff
@@ -1111,9 +1111,8 @@ def create_task_decoder(task_info, g, decoder_input_dim, train_task):
     elif task_info.task_type in [BUILTIN_TASK_LINK_PREDICTION]:
         return create_builtin_lp_decoder(g, decoder_input_dim, task_info.task_config, train_task)
     elif task_info.task_type in [BUILTIN_TASK_RECONSTRUCT_NODE_FEAT]:
-        return create_builtin_reconstruct_nfeat_decoder(g, decoder_input_dim, task_info.task_config, train_task)
-    elif task_info.task_type in [BUILTIN_TASK_RECONSTRUCT_EDGE_FEAT]:
-        return create_builtin_reconstruct_efeat_decoder(g, decoder_input_dim, task_info.task_config, train_task)
+        return create_builtin_reconstruct_nfeat_decoder(
+            g, decoder_input_dim, task_info.task_config, train_task)
```
Comment (classicsong) [inline python/graphstorm/gsf.py:1115]: Why remove the edge feature reconstruction?

**Comment by jalencato** [comment]
May you check if the error in the CI is related to any code change here?

File: python/graphstorm/gsf.py (hunk 2 above)
Comment (thvasilo) [reply python/graphstorm/gsf.py:1115]: Ah good catch I'll add this back.

**Review by classicsong — APPROVED** [review]
LGTM

**Pull request merged**